### PR TITLE
Tower of Hanoi in Java

### PR DESCRIPTION
**Pull request by khusheekapoor**
Hey. This is my code to solve the Tower of Hanoi using Java (issue number #828). Please merge this PR. This will be my first PR. Thanks!

### CODE REVIEW

File: 828-Tower-of-Hanoi-Java.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    
+   public static void main()
+   {
+       Tower_of_Hanoi ob = new Tower_of_Hanoi();
+       System.out.println("Enter number of discs:");
+       Scanner scanner = new Scanner(System.in);
+       int N = scanner.nextInt();
+       ob.Shift(N,"A","B","C");
+    }
+}
```
Comment (sayantani11) [inline 828-Tower-of-Hanoi-Java.txt:43]: Add sample input and sample output

**Review by sayantani11 — CHANGES_REQUESTED** [review]

File: 828-Tower-of-Hanoi-Java.txt (hunk 1 above)
Comment (khusheekapoor) [reply 828-Tower-of-Hanoi-Java.txt]: Done, thank you for your guidance. Really appreciated by a beginner :)

**Comment by sayantani11** [comment]
@khusheekapoor I am a beginner as well....This is a community where we help each other grow and learn new things everyday :)

**Review by raksha009 — COMMENTED** [review]
Plz go through the guidelines and link the appropriate folder.

**Review by sayantani11 — APPROVED** [review]

**Comment by khusheekapoor** [comment]
@raksha009 Are any more changes required? If not, could you please merge my PR? Eagerly waiting!

**Comment by raksha009** [comment]
> Plz go through the guidelines and link the appropriate folder.

Plz go through pinned issues and guidelines!

**Comment by khusheekapoor** [comment]
> > Plz go through the guidelines and link the appropriate folder.
> 
> Plz go through pinned issues and guidelines!



> > Plz go through the guidelines and link the appropriate folder.
> 
> Plz go through pinned issues and guidelines!

@raksha009 I have proposed changes in the algorithms -> recursion -> tower of hanoi in java. 
What else should I do?

**Comment by raksha009** [comment]
> @raksha009 I have proposed changes in the algorithms -> recursion -> tower of hanoi in java.
> What else should I do?

Dear, your changes aren't reflected here. Have a look at the files changed. It shows only yr code is added but not the updated readme of recursion.

**Comment by khusheekapoor** [comment]
> > @raksha009 I have proposed changes in the algorithms -> recursion -> tower of hanoi in java.
> > What else should I do?
> 
> Dear, your changes aren't reflected here. Have a look at the files changed. It shows only yr code is added but not the updated readme of recursion.

@raksha009 , am I on the correct page now?

File: Code/Java/Tower_of_Hanoi.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
+
+/**
```
Comment (raksha009) [inline Code/Java/Tower_of_Hanoi.java:2]: add comments like 
/*

Do not add * in between the comments

File: Code/Java/Tower_of_Hanoi.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
+
```
Comment (raksha009) [inline Code/Java/Tower_of_Hanoi.java:1]: Remove this line

File: Code/Java/Tower_of_Hanoi.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+/**
+ * Tower of Hanoi. Recursive method is used.
+ * Author - Khushee Kapoor
```
Comment (raksha009) [inline Code/Java/Tower_of_Hanoi.java:4]: Being in open source, we make contributions to help wider community. So, plz remove this line :)

File: Code/Java/Tower_of_Hanoi.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+   public static void main()
+   {
+       Tower_of_Hanoi ob = new Tower_of_Hanoi();
+       System.out.println("Enter number of discs:");
+       Scanner scanner = new Scanner(System.in);
+       int N = scanner.nextInt();
+       ob.Shift(N,"A","B","C");
+    }
+}
+/** Sample Input: 3
```
Comment (raksha009) [inline Code/Java/Tower_of_Hanoi.java:50]: Follow format
Test Cases
Input
Output

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Code/Java/Tower_of_Hanoi.java (hunk 2 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java]: Changes made

File: Code/Java/Tower_of_Hanoi.java (hunk 3 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java]: Changes made

File: Code/Java/Tower_of_Hanoi.java (hunk 4 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java]: Changes made

File: Code/Java/Tower_of_Hanoi.java (hunk 5 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java]: Changes made

**Comment by khusheekapoor** [comment]
Also, @raksha009 how do I squash my commits (is it required?)

File: Code/Java/Tower_of_Hanoi.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
+/* Tower of Hanoi. Recursive method is used.
```
Comment (raksha009) [inline Code/Java/Tower_of_Hanoi.java:1]:
```suggestion
/* 
Tower of Hanoi - Recursive method is used.
```

File: Code/Java/Tower_of_Hanoi.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  2. [Recurisve Case]
+       Shift(N-1,A,C,B)
+       Display("Moving Disc A -> C")
+       Shift(N-1,B,A,C)
+  So first recursive call moves n-1 disks from ‘first’ to ‘middle’ using ‘last’. So after that call 
+  n-1 disks are in ‘middle ‘ peg in order of size and the ‘first’ peg contains the nth disk i.e, 
+  the largest one. So, now move that disk from ‘first’ peg to ‘last’ peg. Then again by the 
+  2nd recursive call move n-1 disk from ‘middle’ peg to ‘last’ peg using ‘first’ peg. So , after 
+  all these, ‘last’ peg contains all disks in order of size . */
+import java.util.*;
```
Comment (raksha009) [inline Code/Java/Tower_of_Hanoi.java:19]: Add a new line here

File: Code/Java/Tower_of_Hanoi.java [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+public class Tower_of_Hanoi
+{
+   public void Shift(int N, String first, String middle, String last)
+   {
+       if (N==1)
+       { 
+           System.out.println("Disc moved from \t"+first+"->"+last);
```
Comment (raksha009) [inline Code/Java/Tower_of_Hanoi.java:26]: Add space in between operators like ==, < , > etc

File: Code/Java/Tower_of_Hanoi.java [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+   {
+       Tower_of_Hanoi ob = new Tower_of_Hanoi();
+       System.out.println("Enter number of discs:");
+       Scanner scanner = new Scanner(System.in);
+       int N = scanner.nextInt();
+       ob.Shift(N,"A","B","C");
+    }
+}
+/* Test Cases-
+  Input: 3
```
Comment (raksha009) [inline Code/Java/Tower_of_Hanoi.java:46]:
```suggestion
 1.
  Input: 3
```

File: Code/Java/Tower_of_Hanoi.java [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+  Output:
+  Disc moved from 	A->C
+  Disc moved from 	A->B
+  Disc moved from 	C->B
+  Disc moved from 	A->C
+  Disc moved from 	B->A
+  Disc moved from 	B->C
+  Disc moved from 	A->C
+  Input: 4
```
Comment (raksha009) [inline Code/Java/Tower_of_Hanoi.java:55]:
```suggestion
  2.
  Input: 4
```
Add a new line here first

File: Code/Java/Tower_of_Hanoi.java [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+  Disc moved from 	A->C
+  Disc moved from 	B->C
+  Disc moved from 	B->A
+  Disc moved from 	C->A
+  Disc moved from 	B->C
+  Disc moved from 	A->B
+  Disc moved from 	A->C
+  Disc moved from 	B->C */
+
+/* Time Complexity: O(2^n)
```
Comment (raksha009) [inline Code/Java/Tower_of_Hanoi.java:73]: Include this in same comment. Specify n

**Review by raksha009 — CHANGES_REQUESTED** [review]
Add comments above functions and in between the code where ever required

**Comment by khusheekapoor** [comment]
Hey @raksha009 I have made the changes. Please review further. Thanks!

File: Code/Java/Tower_of_Hanoi.java (hunk 7 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java:21]: Changes made.

File: Code/Java/Tower_of_Hanoi.java (hunk 6 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java]: Changes made.

File: Code/Java/Tower_of_Hanoi.java (hunk 7 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java:21]: Changes made.

File: Code/Java/Tower_of_Hanoi.java (hunk 8 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java:31]: Changes made.

File: Code/Java/Tower_of_Hanoi.java (hunk 9 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java:60]: Changes made.

File: Code/Java/Tower_of_Hanoi.java (hunk 10 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java:71]: Changes made.

File: Code/Java/Tower_of_Hanoi.java (hunk 11 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java]: Changes made.

File: Code/Java/Tower_of_Hanoi.java [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+   {
+       //recursive function
+       if (N == 1)
+       { 
+           //base case
+           System.out.println("Disc moved from \t"+first+"->"+last);
+       }
+       else
+       {
+           /*
```
Comment (raksha009) [inline Code/Java/Tower_of_Hanoi.java:35]: Do not add multiline comment in between the file. Add this in single line comments

File: Code/Java/Tower_of_Hanoi.java [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+         
+           Shift(N-1, first, last, middle);
+           System.out.println("Disc moved from \t"+first+"->"+last);
+           Shift(N-1, middle, first, last);
+        }
+    }
+    
+   public static void main()
+   {
+       //driver fucntion
```
Comment (raksha009) [inline Code/Java/Tower_of_Hanoi.java:50]: Remove this comment

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Code/Java/Tower_of_Hanoi.java (hunk 12 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java]: Made the changes.

File: Code/Java/Tower_of_Hanoi.java (hunk 13 above)
Comment (khusheekapoor) [reply Code/Java/Tower_of_Hanoi.java]: Made the changes.

**Review by raksha009 — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**